### PR TITLE
Add Replace

### PR DESCRIPTION
**Pull request by jdtzmn**
This PR adds the upcoming AVM 7 `replace2` and `replace3` opcodes.

### CODE REVIEW

File: pyteal/ir/ops.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from dataclasses import replace
```
Comment (barnjamin) [inline pyteal/ir/ops.py:1]: intellisense got too excited?

File: pyteal/__init__.pyi [hunk 2]
```diff
@@ -138,6 +138,7 @@ __all__ = [
     "Or",
     "Pop",
     "Reject",
+    "Replace",
```
Comment (michaeldiamant) [inline pyteal/__init__.pyi:141]: @jdtzmn Unrelated to line:  While reviewing the PR, I added a couple of tests in https://github.com/jdtzmn/pyteal/pull/1.

Can you see if you feel it's worth pulling into the PR?  Otherwise, we can close the child PR.
